datamade/how-to
# Add Python and JavaScript linting to new Django app template

## Overview

See title.

Handles #91, handles #115

### Notes

I adding linting to the testing step, which brings this in line with our Gatsby template. In the future, it may be a good idea to investigate [hooking linters up to pre-commit hooks](https://flake8.pycqa.org/en/latest/user/using-hooks.html) so misformatted code never makes its way into version control. I'll leave that for a separate work effort, though.

## Testing Instructions

* Pull down this branch.
* Follow the instructions to create a new Django app from the `new-django-app` template, accepting all the defaults.
* Switch into your new app directory: `cd /path/to/my-new-app`
* Run the tests and confirm that you see no output from either linting step, and that the `pytest` tests pass: `docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app`
